cpp/src/lib/osd_access.cc

```diff
@@ -61,14 +61,38 @@ void OsdAccess::_remove_ctx(osd_t osd) {
   _osd_ctxs.erase(osd);
 }
 
-void OsdAccess::update(std::vector<std::pair<osd_t, info_caps>> &infos) {
-  ALBA_LOG(DEBUG, "OsdAccess::update");
-  std::lock_guard<std::mutex> lock(_osd_infos_mutex);
 
-  _osd_infos.clear();
-  for (auto &p : infos) {
-    _osd_infos.emplace(p.first, std::move(p.second));
-  }
+void OsdAccess::update(Proxy_client& client) {
+    if(!_filling){
```

Comment:
you may need to specify that you want to do a 'volatile' read here?
(that's the c# terminology, in c++ the stuff in std::memory_order might be related)
@redlicha can you review this part?

nit: please do a clang-format on this file.